components/cache/cache_invalidation.rst

```diff
@@ -0,0 +1,107 @@
+several cached items, keeping them in sync can be difficult.
+
+The Symfony Cache component provides three mechanisms to help solve this problem:
+
+* Tags based invalidation for managing data dependencies;
+* Namespace based invalidation for context and sub-contexts dependend data;
+* Expiration based invalidation for time related dependencies.
+
+.. versionadded:: 3.2
+    Tags and namespace based invalidation was introduced in Symfony 3.2.
+
+Using Cache Tags
+----------------
+
+To benefit from the tags based invalidation, your responsiblity is to attach the
+proper tags to each cached items. Each tag is a plain string identifier that you
+can use to trigger the removal of all items that had this tag attached to them.
+
+To attach tags to cached items, you need to use the
+:method:`Symfony\\Component\\Cache\\CacheItem::tag` method that is implemented by
+cache items, as returned by cache adapters::
+
+    $item = $cache->getItem('cache_key');
+    // ...
+    // add one or more tags
+    $item->tag('tag_1');
+    $item->tag(array('tag_2', 'tag_3'));
+    $cache->save($item);
+
+If `$cache` implements :class:`Symfony\\Component\\Cache\\TagAwareAdapterInterface`,
+you can invalidate the cached items by calling
+:method:`Symfony\\Component\\Cache\\TagAwareAdapterInterface::invalidateTags`::
+
+    // invalidate all items related to `tag_1`
+    $cache->invalidateTags('tag_2');
+
+    // or invalidate all items related to `tag_1` or `tag_3`
+    $cache->invalidateTags(array('tag_1', 'tag_3'));
+
+Of course when you know the cache key, it's better to call
+``$cache->deleteItem('cache_key');``, but this key is sometimes hard to track
+and this is where cache tags become useful.
+
+Tag Aware Adapters
+~~~~~~~~~~~~~~~~~~
+
+To store tags, you need to wrap a cache adapter with the
+:class:`Symfony\\Component\\Cache\\Adapter\\TagAwareAdapter` class or implement
+:class:`Symfony\\Component\\Cache\\Adapter\\TagAwareAdapterInterface` and its only
+:method:`Symfony\\Component\\Cache\\Adapter\\TagAwareAdapterInterface::invalidateTags`
+method by yourself.
+
+The :class:`Symfony\\Component\\Cache\\Adapter\\TagAwareAdapter` class implements
+instantaneous invalidation. It needs one or two cache adapters: the required one
+is used to store cached items; the second one is used to store tags and their
+invalidation version number (conceptually similar to their latest invalidation
+date). When only one adapter is used, items and tags are all stored in the same
+place. By using two adapters, you can e.g. store some big cached items on the
+filesystem or in the database, and keep tags in a Redis database to sync all your
+fronts and have very fast invalidation checks::
+
+    $cache = new TagAwareAdapter(
+        // Adapter for cached items
+        new FilesystemAdapter(),
+        // Adapter for tags
+        new RedisAdapter('redis://localhost')
+    );
+
+Using Cache Namespaces
+----------------------
+
+By using adapters that implement the
```

Comment:
this section is missing some information, at least I can't understand what it's explaining.

 * What are cache namespaces?
 * What is forking?
 * What are "context-dependent variations of your cached items"?
 * When would I use cache namespaces and forks?